deepchecks/core/check.py

```diff
@@ -206,6 +212,79 @@ def _display_to_json(self) -> List[Tuple[str, str]]:
+        """
+        assert wandb, 'Missing wandb dependency, please install wandb'
+        check_metadata = self._get_metadata()
+        if wandb_init:
+            kwargs['project'] = kwargs.get('project', check_metadata['name'])
+            kwargs['project'] = kwargs.get('config', check_metadata)
+            wandb.init(**kwargs)
+            section_suffix = ''
+        else:
+            section_suffix = check_metadata['name'] + '/'
+        if self.conditions_results:
+            cond_df = get_conditions_table([self], icon_html=False)
+            cond_table = wandb.Table(dataframe=cond_df.data, allow_mixed_types=True)
+            wandb.log({f'{section_suffix}conditions_table': cond_table}, commit=False)
+        if isinstance(self.value, pd.DataFrame):
+            value = self.value.to_json()
+        elif isinstance(self.value, Styler):
+            value = self.value.data.to_json()
+        elif isinstance(self.value, np.ndarray):
+            value = self.value.tolist()
+        else:
+            value = jsonpickle.dumps(self.value)
+        table_i = 0
+        plot_i = 0
+        old_backend = matplotlib.get_backend()
+        for item in self.display:
+            if isinstance(item, Styler):
+                # print(check_metadata['header'])
+                # print(item.data)
+                wandb.log({f'{section_suffix}display_table_{table_i}':
+                           wandb.Table(dataframe=item.data, allow_mixed_types=True)}, commit=False)
+                table_i += 1
+            elif isinstance(item, pd.DataFrame):
+                wandb.log({f'{section_suffix}display_table_{table_i}':
+                           wandb.Table(dataframe=item, allow_mixed_types=True)}, commit=False)
+                table_i += 1
+            elif isinstance(item, str):
+                pass
+            elif isinstance(item, BaseFigure):
+                wandb.log({f'{section_suffix}plot_{plot_i}': wandb.Plotly(item)})
+                plot_i += 1
+            elif callable(item):
+                try:
+                    matplotlib.use('Agg')
+                    item()
+                    wandb.log({f'{section_suffix}plot_{plot_i}': plt})
+                    plot_i += 1
+                except Exception:
+                    pass
+            else:
+                matplotlib.use(old_backend)
+                raise Exception(f'Unable to process display for item of type: {type(item)}')
+
+        matplotlib.use(old_backend)
+        data = [check_metadata['header'],
+                str(check_metadata['params']),
+                check_metadata['summary'],
+                value]
+        final_table = wandb.Table(columns=['header', 'params', 'summary', 'value'])
+        final_table.add_data(*data)
+        wandb.log({f'{section_suffix}results': final_table}, commit=False)
+        print('logged: ' + check_metadata['header'])
+        if wandb_init:
+            wandb.finish()
```

Comment:
If we're gonna use it like this, we need to make it clear that the flag wandb_init actually means we're gonna create a run that will only exist for deepchecks, and will exit when this function exits. If so, we need to change the name of this flag, e.g. to "output_to_dedicated_run" or something.